pygmt/base_plotting.py

```diff
@@ -1922,3 +1922,240 @@ def update_pointers(data_pointers):
+
+                    **1**,\\ **2**:
+                    longitude, latitude, of station (**-:** option interchanges
+                    order)
+                    **3**:
+                    eps1, the most extensional eigenvalue of strain tensor,
+                    with extension taken positive.
+                    **4**:
+                    eps2, the most compressional eigenvalue of strain tensor,
+                    with extension taken positive.
+                    **5**:
+                    azimuth of eps2 in degrees CW from North.
+
+
+        Other Parameters
+        ----------------
+        vector : bool or str
+            Modify vector parameters. For vector heads, append vector head size
+            [Default is 9p]. For specifying additional attributes, see
+            :gmt-docs:`supplements/geodesy/velo.html#vector-attributes`.
+
+        {B}
+
+        {CPT}
+
+        rescale : str
+            can be used to rescale the uncertainties of velocities
+            (``scaling='e'`` and ``scaling='r'``) and rotations
+            (``scaling='w'``). Can be combined with the confidence variable.
+
+        uncertainty_color : str
+            Sets the color or shade used for filling uncertainty wedges
+            (``scaling='w'``) or velocity error ellipses (``scaling='e'`` or
+            ``scaling='r'``). [If *uncertainty_color* is not specified, the
+            uncertainty regions will be transparent]. More details on
+            :gmt-docs:`cookbook/features.html#gfill-attrib`.
+
+        facecolor : str
+            Select color or pattern for filling of symbols or polygons
+            [Default is no fill]. More details on
+            :gmt-docs:cookbook/features.html#gfill-attrib`.
+
+        line: str
+            Draw lines. Ellipses and fault planes will have their outlines
+            drawn using current pen (see *pen*).
+
+        no_clip: str
+            Do NOT skip symbols that fall outside the frame boundary specified
+            by *region*. [Default plots symbols inside frame only].
+
+        pen : str
+            Set pen attributes for velocity arrows, ellipse circumference and
+            fault plane edges. [Defaults: width = default, color = black,
+            style = solid].
+
+        {U}
+        {V}
+        {XY}
+        {p}
+        {t}
+        """
+        kwargs = self._preprocess(**kwargs)
+
+        kind = data_kind(data)
+
+        extra_arrays = []
+        if "G" in kwargs and not isinstance(kwargs["G"], str):
+            if kind != "vectors":
+                raise GMTInvalidInput(
+                    "Can't use arrays for color if data is matrix or file."
+                )
+            extra_arrays.append(kwargs["G"])
+            del kwargs["G"]
+
+        if scaling is not None:
+            if not isinstance(scaling,str):
+                raise GMTInvalidInput(
+                    "scaling has to be a string."
+                )
+            extra_arrays.append(scaling)
```

Comment:
We'll need to change this code here since 'vector' style inputs (e.g. `fig.velo(x=..., y=..., color=...)`) isn't supported, only `matrix` (pandas.DataFrame) and `str` (filename). The `extra_arrays` variable isn't actually used and can be removed.

Reply:
By changing, you mean removing the lines 2129 to 2136?

Reply:
Yes :)